tests/sharedv4/sharedv4_test.go

```diff
@@ -0,0 +1,269 @@
+						fmt.Sprintf("stop volume driver %s on app %s's node: %s",
+							Inst().V.String(), ctx.App.Key, attachedNode.Name),
+						func() {
+							StopVolDriverAndWait([]node.Node{*attachedNode})
+						})
+
+					Step(
+						fmt.Sprintf("starting volume %s driver on app %s's node %s",
+							Inst().V.String(), ctx.App.Key, attachedNode.Name),
+						func() {
+							StartVolDriverAndWait([]node.Node{*attachedNode})
+						})
+
+					Step("Giving few seconds for volume driver to stabilize", func() {
+						time.Sleep(20 * time.Second)
+					})
+
+					Step(fmt.Sprintf("validate app %s", attachedNode.Name), func() {
+						ValidateContext(ctx)
+					})
+				}
+			}
+		})
+
+	})
+
+})
+
+// This test performs sharedv4 nfs server pod termination failover use case
+var _ = Describe("{NfsServerKill}", func() {
+	var contexts []*scheduler.Context
+
+	It("has to validate that the new pods started successfully after nfs server node is terminated", func() {
+		contexts = make([]*scheduler.Context, 0)
+		nodesToKill := []node.Node{}
+		var err error
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("nfskill-%d", i))...)
+		}
+
+		ValidateApplications(contexts)
+		for _, ctx := range contexts {
+			var appVolumes []*volume.Volume
+			Step(fmt.Sprintf("get volumes for %s app", ctx.App.Key), func() {
+				appVolumes, err = Inst().S.GetVolumes(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(appVolumes).NotTo(BeEmpty())
+			})
+			for _, v := range appVolumes {
+
+				Step("get attached node and stop the instance", func() {
+					currNodes := node.GetStorageDriverNodes()
+					countOfCurrNodes := len(currNodes)
+
+					attachedNode, err := Inst().V.GetNodeForVolume(v, defaultCommandTimeout, defaultCommandRetry)
+					nodesToKill = append(nodesToKill, *attachedNode)
+
+					// Reboot node and check driver status
+					Step(fmt.Sprintf("stop node : %v having volume: %v attached", attachedNode.Name, v.Name), func() {
+
+						KillANodeAndValidate(nodesToKill)
+
+						Step("validate applications", func() {
+							for _, ctx := range contexts {
+								ValidateContext(ctx)
+							}
+						})
+
+						Step(fmt.Sprintf("validate node: %v is stopped", attachedNode.Name), func() {
+							currNodes = node.GetStorageDriverNodes()
+							isNodePresent := false
+							for _, currNode := range currNodes {
+								if currNode.Name == attachedNode.Name {
+									isNodePresent = true
+									break
+								}
+							}
+							Expect(isNodePresent).To(BeFalse())
+						})
+
+						Step(fmt.Sprintf("wait to new instance to start scheduler: %s and volume driver: %s",
+							Inst().S.String(), Inst().V.String()), func() {
+							time.Sleep(2 * time.Minute)
+							currNodes = node.GetStorageDriverNodes()
+							Expect(countOfCurrNodes).To(Equal(len(currNodes)))
+							for _, n := range currNodes {
+
+								err = Inst().S.IsNodeReady(n)
+								Expect(err).NotTo(HaveOccurred())
```

Comment:
How can we be sure that this will not take more than 2 minutes? Ideally, we should check if the node is ready in a loop with a short sleep interval and a long timeout. But I may be misunderstanding what `IsNodeReady()` does.

Reply:
Yes, IsNodeReadyhas that logic. Here we are waiting 2 mins to start the verification